File: PH2007SDK/developpers/CSTemplate/myPlayer.cs
/* Template done by
 * 
 * Richard Clark
 * Project Hoshimi Lead Manager
 * Contact at ori@c2i.fr
 * 
*/
using System;
using System.Collections.Generic;
using System.Text;
using PH.Common;
using PH.Map;
using System.Drawing;

namespace CSTemplate
{
    public class myPlayer : PH.Common.Player
    {
        #region Initializations
        public myPlayer() { }
        public myPlayer(string name, int ID)
            : base(name, ID)
        {
            this.ChooseInjectionPoint += new PH.Common.ChooseInjectionPointHandler(myPlayer_ChooseInjectionPoint);
            this.WhatToDoNext += new PH.Common.WhatToDoNextHandler(myPlayer_WhatToDoNext);
        }
        public override System.Drawing.Bitmap Flag
        {
            get { return Properties.Resources.rcFlag; }
        } 
        #endregion

        public enum WhatToDoNextAction
        {
            NothingToDo = 0,
        }
        private WhatToDoNextAction m_WhatToDoNext = WhatToDoNextAction.NothingToDo;
        public WhatToDoNextAction AI_WhatToDoNext
        {
            get { return m_WhatToDoNext; }
            set { m_WhatToDoNext = value; }
        }
        
        private void UpdateInformations()
        {}

        private void AI_DoActions()
        {
            switch (this.AI_WhatToDoNext)
            {
                case WhatToDoNextAction.NothingToDo:
                    break; 
            }
        }

        private void myPlayer_WhatToDoNext()
        {
            UpdateInformations();

            if (this.AI.State == NanoBotState.WaitingOrders)
                AI_DoActions();

            foreach (NanoBot bot in this.NanoBots)
                if (bot is IActionable && bot.State == NanoBotState.WaitingOrders)
                    ((IActionable)bot).DoActions();
        }

        #region Injection Point
        private List<Entity> m_AznEntities = new List<Entity>();
        private List<Entity> m_HoshimiEntities = new List<Entity>();
        public List<Entity> AZNEntities { get { return m_AznEntities; } }
        public List<Entity> HoshimiEntities { get { return m_HoshimiEntities; } }
        
        private void myPlayer_ChooseInjectionPoint()
        {
            //storing AZN and Hoshimi Points
            foreach (Entity ent in this.Tissue.Entities)
            {
                switch (ent.EntityType)
                {
                    case EntityEnum.AZN:
                        AZNEntities.Add(ent);
                        break;
                    case EntityEnum.HoshimiPoint:
                        HoshimiEntities.Add(ent);
                        break;
                }
            }

            //CHANGE THIS LINE
            this.InjectionPointWanted = new Point(100, 100);
        } 
        #endregion

    }
}

File: PH2007SDK/developpers/VBTemplate/myPlayer.vb
'Template done by
' 
'Richard Clark
'Project Hoshimi Lead Manager
'Contact at ori@c2i.fr
' 
Imports PH.Common
Imports PH.Map
Imports System.Drawing

Public Class myPlayer
    Inherits PH.Common.Player

#Region "Initializations"
    Public Sub New()
    End Sub

    Public Sub New(ByVal name As String, ByVal ID As Integer)
        MyBase.New(name, ID)
    End Sub

    Public Overrides ReadOnly Property Flag() As System.Drawing.Bitmap
        Get
            Return My.Resources.rcFlag
        End Get
    End Property
#End Region

    Public Enum WhatToDoNextAction
        NothingToDo = 0
    End Enum


    Private m_WhatToDoNext As WhatToDoNextAction = WhatToDoNextAction.NothingToDo
    Public Property AI_WhatToDoNext() As WhatToDoNextAction
        Get
            Return m_WhatToDoNext
        End Get
        Set(ByVal value As WhatToDoNextAction)
            m_WhatToDoNext = value
        End Set
    End Property

    Private Sub UpdateInformations()
    End Sub
    Private Sub AI_DoActions()

        Select Case Me.AI_WhatToDoNext
            Case WhatToDoNextAction.NothingToDo
        End Select
    End Sub

    Private Sub myPlayer_WhatToDoNextEvent() Handles Me.WhatToDoNext
        UpdateInformations()

        If Me.AI.State <> NanoBotState.WaitingOrders Then AI_DoActions()

        For Each bot As NanoBot In Me.NanoBots
            If TypeOf (bot) Is IActionable AndAlso bot.State = NanoBotState.WaitingOrders Then
                CType(bot, IActionable).DoActions()
            End If
        Next
    End Sub

#Region "Injection Point"
    Private m_AZNEntities As New List(Of Entity)
    Private m_HoshimiEntities As New List(Of Entity)
    Public ReadOnly Property AZNEntities() As List(Of Entity)
        Get
            Return m_AZNEntities
        End Get
    End Property
    Public ReadOnly Property HoshimiEntities() As List(Of Entity)
        Get
            Return m_HoshimiEntities
        End Get
    End Property

    Private Sub myPlayer_ChooseInjectionPointEvent() Handles Me.ChooseInjectionPoint
        'storing AZN and Hoshimi Points
        For Each ent As Entity In Me.Tissue.Entities
            Select Case ent.EntityType
                Case EntityEnum.AZN
                    AZNEntities.Add(ent)
                Case EntityEnum.HoshimiPoint
                    HoshimiEntities.Add(ent)
            End Select
        Next

        'CHANGE THIS LINE
        Me.InjectionPointWanted = New Point(100, 100)
    End Sub
#End Region


End Class
